app/javascript/controllers/move_procedures_position_controller.ts

```diff
@@ -0,0 +1,135 @@
+import { ApplicationController } from './application_controller';
+
+export class MoveProceduresPositionController extends ApplicationController {
+  connect() {
+    const { firstCard, lastCard } = this.selectFirstAndLastCards();
+    this.updateButtonsStates(firstCard, null, 'Up');
+    this.updateButtonsStates(lastCard, null, 'Down');
+  }
+
+  moveUp(event: Event) {
+    event.preventDefault();
+    const button = event.currentTarget as HTMLButtonElement;
+    const upCard = button.closest('.fr-card');
+    const { firstCard, lastCard } = this.selectFirstAndLastCards();
+    if (upCard!.previousElementSibling === firstCard) {
+      const part = 'Up';
+      this.updateButtonsStates(upCard, firstCard, part);
+    }
+    if (upCard === lastCard) {
+      const part = 'Down';
+      this.updateButtonsStates(upCard!.previousElementSibling, upCard, part);
+    }
+    this.switchCards(upCard!, upCard!.previousElementSibling!).then(() => {
+      upCard!.parentNode!.insertBefore(upCard!, upCard!.previousElementSibling);
+      this.updateInputsValues(
+        upCard!,
+        upCard!.previousElementSibling,
+        upCard!.nextElementSibling
+      );
+    });
+  }
+
+  moveDown(event: Event) {
+    event.preventDefault();
+    const button = event.currentTarget as HTMLButtonElement;
+    const downCard = button.closest('.fr-card');
+    const { firstCard, lastCard } = this.selectFirstAndLastCards();
+    if (downCard === firstCard) {
+      const part = 'Up';
+      this.updateButtonsStates(downCard!.nextElementSibling, downCard, part);
+    }
+    if (downCard!.nextElementSibling === lastCard) {
+      const part = 'Down';
+      this.updateButtonsStates(downCard, downCard!.nextElementSibling, part);
+    }
+    this.switchCards(downCard!.nextElementSibling!, downCard!).then(() => {
+      downCard!.parentNode!.insertBefore(
+        downCard!.nextElementSibling!,
+        downCard
+      );
+      this.updateInputsValues(
+        downCard!,
+        downCard!.previousElementSibling,
+        downCard!.nextElementSibling
+      );
+    });
+  }
+
+  private async switchCards(upCard: Element, downCard: Element): Promise<void> {
```

Comment:
:trollface: on est pas studio gibli :trollface: je propose dropper les animations, bcp plus simple.

[0001-poc-refactor-no-animations-we-are-not-here-to-do-tha.patch](https://github.com/user-attachments/files/18035869/0001-poc-refactor-no-animations-we-are-not-here-to-do-tha.patch)

Reply:
suite à un echange ac Marlene, bravo pr les animations ;-)